cpp/src/io/csv/csv_reader.cu

```diff
@@ -197,6 +198,84 @@ __device__ void setBit(gdf_valid_type* address, int bit) {
 		return base_padding + num_columns * column_bytes; 
 	}
 }
+/**
+* @brief Removes the first and Last quote in the string
+*/
+string removeQoutes(string str, char quotechar) {
+	// Exclude first and last quotation char
+	const size_t first_quote = str.find(quotechar);
+	if (first_quote != string::npos) {
+		str.erase(first_quote, 1);
+	}
+	const size_t  last_quote = str.rfind(quotechar);
+	if (last_quote != string::npos) {
+		str.erase(last_quote, 1);
+	}
+
+	return str;
+}
+
+/**
+ * @brief Parse the first row to set the column names in the raw_csv parameter 
+ *
+ * The first row can be either the header row, or the first data row
+ *
+ * @param[in,out] raw_csv Structure containing the csv parsing parameters
+ * and intermediate results
+ * 
+ * @return gdf_error with error code on failure, otherwise GDF_SUCCESS
+*/
+gdf_error setColumnNamesFromCsv(raw_csv_t* raw_csv) {
+	vector<char> first_row = raw_csv->header;
```

Comment:
I like moving this into a separate function.  One suggestion I have is to add this `first_row` into the `raw_csv` structure and initialize it in the function `uploadDataToDevice` where `raw_csv->header` is being initialized.  That way we don't need to copy memory back to host and you can save the whole if statement below.